### ISSUE TITLE
Feedback for Mini-Exercise 6 by Magnus

### ISSUE THREAD

**Issue by MagnusJMJ** [Epsilon99/AestheticProgrammingCourse#9]
I really like the program - it's a great example of generativity in software, and the sprites are a _really_ nice touch! I got a good chuckle out of the `spriteHell()` function, but I'd probably have done it the same way. 

The two-dimensional array is also a vast improvement over the system you used in your Game of Life sketch (which I used too, and now I can never look at it the same way again).

I wanted to ask though; why do you declare a class for the dungeon tiles? Does it afford something that a "regular" constructor function doesn't? I've never used classes myself, but it seems less versatile than constructor functions (because classes are not hoisted, and the body of a class is run in strict mode)

On another note, big ups for descriptive variable names and lots of whitespace - the only problem I had was that in `function draw()` the nested `if`/`else` statements become a little hard to follow without an indentation guide.

// Magnus

**Comment by Epsilon99** [comment]
Hi Magnus,

Thanks for your feedback, I appreciate it.

The two-dimensional array really eases up the process of Game of Life, so maybe have a look at it. I do think I send you a link to a tutorial, but I'll throw it in [here for the sake of context](https://gamedevelopment.tutsplus.com/tutorials/generate-random-cave-levels-using-cellular-automata--gamedev-9664).

The use of either constructor or classes in this case is probably not too different from each other. I have a lot of 'bad habbits' when it comes to coding. I'm used to work with C# and unity, where a constructor function is not really a thing, as far as I know, but instead you create classes whenever you need'em.
  However, I want to point out that a class can have a sub-class. This is where the power of OOP (Object-Oriented Programming) shines. Basicly means that I can have a class called car, which has all the properties of car in it, but I could also have a subclass which is the brand of the car, where I declare everything about the cars more specific details. Now you have a more complex and deeper way of structurering your classes and thus having more flexability when handling greater objects. 
  But, as I said, in this case I could probably have used either :)

Regarding the draw() loop, I do realize I lacked a bit of explination. The thing is, statemachines, which this clasifies at, I find it easier to explain  these with a statemachine-diagram, rather then commenting. But I do agree that it's messy and I could have written some comments to help guide you throu it.

Again, a lot of corners were cut to reach the deadline, including the spriteHell() :)
